### DISCUSSION TITLE
Check if a file exists through the link

### DISCUSSION THREAD

**Discussion by e5pe** [SilentVoid13/Templater#150] [Help] [answered]
Is there any way to check if a file exists throughthe link? Like a function in the [File Module](https://silentvoid13.github.io/Templater/docs/internal-variables-functions/internal-modules/file-module).

For exaple I have this template for the daily notes:

`[[<% tp.date.now("DD-MM-YYYY", -1) %>]]`

But I just want to create it if that reference has a file already because it generates a date for yesterday. 

Thank you.

**Comment by SilentVoid13** [comment]
Hey @e5pe, if I understand correctly, you want something like `tp.file.exist("[[MyFile]]")` ?

**Reply by e5pe** [reply]
Something like that yeah. Does that function exist?

**Reply by SilentVoid13** [reply]
It does not yet exist, but it's a good idea, I'll probably add this later.

**Comment by SilentVoid13** [comment, marked as answer]
This has been implemented in Templater 1.6.0, update to try it!

**Reply by e5pe** [reply]
I saw it, thank you!

**Comment by stefanku** [comment]
Would it be possible to create a similar functions for _folders_ as well?

**Reply by shabegom** [reply]
This should work for now:

`<% app.vault.adapter.exists("/Folder", true) %>`

**Reply by stefanku** [reply]
Wonderful - thank you!
